src/main/java/domain/Ladder.java

```diff
@@ -1,16 +1,46 @@
 package domain;
 
+import util.BooleanGenerator;
+
 import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 
 public class Ladder {
 
     private final List<Line> lines;
 
-    public Ladder(List<Line> lines) {
-        this.lines = lines;
+    public Ladder(Height height, Weight weight, BooleanGenerator booleanGenerator) {
+        this.lines = generate(height, weight, booleanGenerator);
+    }
+
+    private List<Line> generate(Height height, Weight weight, BooleanGenerator booleanGenerator) {
+        return IntStream.range(0, height.getHeight())
+                        .mapToObj((count) -> new Line(weight.getWeight(), booleanGenerator))
+                        .collect(Collectors.toList());
     }
 
     public List<Line> getLines() {
-        return List.copyOf(lines);
+        return lines;
+    }
+
+    public int getResultIndex(int index) {
+        int resultIndex = index;
+        for (Line line : lines) {
+            resultIndex = getNextIndex(index, line);
```

Comment:
도메인의 중요한 부분이 잘못된 것 같습니다. getNextIndex의 인자로 index를 받고 있어서 매번 같은 인자로 받게 됩니다.

이 부분 때문에 Ladder의 lines 필드를 아래와 같이 넣고 테스트해보면 결과값이 틀리게 나옵니다.

```
// a,b,c,d
// x,100,200,x
// 4
// 정답 : a : 200, b : x, c : 100, d : x

lines.add(new Line(List.of(true, false, false)));  
lines.add(new Line(List.of(false, false, true)));  
lines.add(new Line(List.of(false, false, true)));  
lines.add(new Line(List.of(false, true, false)));
```

<img src="https://user-images.githubusercontent.com/22311557/220902429-8ec66892-9a02-4baa-b2c4-35847c157493.png" width="250">

아래처럼 수정이 필요하지 않을까요?
```suggestion
            resultIndex = getNextIndex(resultIndex, line);
```

Reply:
정말 바보 같은 실수네요... 🥲
해당 부분은 수정해 두었습니다.

테스트 때 잡히지 않는 것을 보니 테스트도 부실했었다는 생각이 듭니다...
이를 해결하기 위해 TestGenerator를 만들어 좀 더 자세한 테스트가 가능하도록 수정했습니다!

Reply:
> 테스트 때 잡히지 않는 것을 보니 테스트도 부실했었다는 생각이 듭니다...

개발자로서 좋은 태도를 가지셨군요 👍 테스트 작성 시, 복잡한 도메인 혹은 비즈니스적으로 중요한 도메인은 가급적 테스트에 많은 노력을 기울이시면 좋을 것 같습니다. 추후 이상적인 테스트는 어떤 것일까 고민하게 될텐데요. (시간적 여유가 있다면 [단위테스트](http://www.yes24.com/Product/Goods/104084175) 라는 책 추천드립니다.)